File: Data Structures & Algorithms/duplicate-integer/submission-4.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        bool found = false;

        for (int i=0 ; i<nums.size() ; i++) {
            for(int j=i+1 ; j<nums.size() ; j++) {
                if (nums[i] == nums[j]) {
                    found = true;
                    break;
                }
            }
            if (found) break;
        }
        return found;
    }
        
};

File: Data Structures & Algorithms/permutations/submission-4.py
class Solution:
    def permute(self, nums: List[int]) -> List[List[int]]:
        res, subset = [], []

        def dfs(i, mask):
            if len(subset) == len(nums):
                res.append(subset.copy())
                return
            for i in range(0, len(nums)):
                if not(mask & (1<<i)):
                    subset.append(nums[i])
                    dfs(i+1, mask|(1<<i))
                    subset.pop()
        
        dfs(0, 0)
        return res;